app/src/main/java/org/groundplatform/android/ui/map/gms/features/LineStringRenderer.kt

```diff
@@ -72,6 +85,90 @@ constructor(private val resources: Resources, private val bitmapUtil: BitmapUtil
       jointType = JointType.ROUND
       zIndex = POLYLINE_Z
     }
+
+    updateTooltipMarker(map, geometry)
     return polyline
   }
+
+  private fun updateTooltipMarker(map: GoogleMap, geometry: LineString) {
+    if (geometry.coordinates.size < 2) {
+      removeTooltipMarker()
+      return
+    }
+
+    val lastVertex = geometry.coordinates.last()
+    val secondLastVertex = geometry.coordinates[geometry.coordinates.size - 2]
+
+    val isPolygonClosed = geometry.isClosed()
+
+    if (isPolygonClosed) {
```

Comment:
Same here - the renderer should only care whether there's a tooltip set on the LineString or not, not the number of vertices; it's up to creator of the LineString to set the correct tooltip (or not).